test/robot/web/client/data.go

```diff
@@ -48,50 +67,39 @@ var (
 			return t.trace.target
 		},
 		enumSrc: func() enum {
-			return itemGetter("{{.id}}", "{{.information.Configuration.Hardware.Name}}")(queryArray("/devices/"))
+			return itemGetter("{{.id}}", machineDisplayTemplate, template.FuncMap{})(queryArray("/devices/"))
 		},
 	}
 	hostDimension = &dimension{
-		name: "traceHost",
+		name: "host",
 		valueOf: func(t *task) Item {
-			return t.trace.host
+			return t.host
 		},
 		enumSrc: func() enum {
-			return itemGetter("{{.id}}", "{{.information.Configuration.Hardware.Name}}")(queryArray("/devices/"))
+			return itemGetter("{{.id}}", machineDisplayTemplate, template.FuncMap{})(queryArray("/devices/"))
 		},
 	}
-	kindDimension = &dimension{
-		name:     "kind",
-		enumData: enum{traceKind, replayKind, reportKind},
+	packageDimension = &dimension{
+		name: "package",
 		valueOf: func(t *task) Item {
-			return t.kind
-		},
-	}
-	gapidApkDimension = &dimension{
-		name: "gapid_apk",
-		valueOf: func(t *task) Item {
-			return t.trace.gapidApk
+			return t.pkg
 		},
 		enumSrc: func() enum {
-			return androidToolFieldMerger("gapid_apk", queryArray("/packages/"))
-		},
-	}
-	gapitDimension = &dimension{
-		name: "gapit",
-		valueOf: func(t *task) Item {
-			return t.trace.gapit
-		},
-		enumSrc: func() enum {
-			return hostToolFieldMerger("gapit", queryArray("/packages/"))
+			return itemGetter("{{.id}}", packageDisplayTemplate, template.FuncMap{"isUserType": isUserType})(queryArray("/packages/"))
 		},
 	}
 
-	dimensions = []*dimension{subjectDimension, targetDimension, hostDimension, kindDimension, gapidApkDimension, gapitDimension}
+	dimensions = []*dimension{kindDimension, subjectDimension, targetDimension, hostDimension, packageDimension}
 )
 
-func itemGetter(idPattern string, displayPattern string) func([]interface{}) enum {
+func isUserType(t reflect.Value) bool {
+	// cannot currently use build.Type_UserType to check the type need to fix that.
+	return t.Kind() == reflect.Float64 && t.Float() == float64(2)
```

Comment:
I need to get the actual value for build.Type_UserType, but cannot import the build package as gopherjs doesn't allow importing packages that use cgo, What should we do in this instance?

Reply:
I assume the cgo code is in the device package?
You could try using the [build constraint](https://golang.org/pkg/go/build/#hdr-Build_Constraints) `// -build js` in the go file that has the `import "C"` line.

Reply:
After looking into this I couldn't get any build constraints to keep GopherJS from trying to build the file that imports the cgo code. One idea I had was to build the proto files into javascript as well, which would not only allow this type of value checking, but would let us use the actual type layouts rather than using text templating and json string to interface mappings to read data. This would probably be another PR though.